buildSrc/src/main/kotlin/otel.javaagent-instrumentation.gradle.kts

```diff
@@ -1,142 +1,23 @@
-import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
-import io.opentelemetry.instrumentation.gradle.bytebuddy.ByteBuddyPluginConfigurator
-
 plugins {
-  id("net.bytebuddy.byte-buddy")
-
-  id("otel.java-conventions")
-  id("otel.shadow-conventions")
+  id("otel.javaagent-testing")
+  id("otel.publish-conventions")
 }
 
-extra["mavenGroupId"] = "io.opentelemetry.javaagent.instrumentation"
-// Shadow is only for testing, not publishing.
-extra["noShadowPublish"] = true
+// TODO(anuraaga): Migrate MuzzlePlugin to kotlin script since Java plugins don't seem to be usable" +
+// with plugins block. It will be good to add a property for bootstrapRuntime to that plugin at the
+// time instead of defining it here and reading it in a non-obvious way.
+apply(plugin = "muzzle")
 
-val skipPublish: Boolean? by extra
-
-if (skipPublish != true) {
-  apply(plugin = "otel.publish-conventions")
-}
-
-apply(plugin = "otel.instrumentation-conventions")
-
-if (projectDir.name == "javaagent") {
-  apply(plugin = "muzzle")
-
-  base.archivesBaseName = projectDir.parentFile.name
-}
+extra["mavenGroupId"] = "io.opentelemetry.javaagent.instrumentation"
 
-val toolingRuntime by configurations.creating {
-  isCanBeConsumed = false
-  isCanBeResolved = true
-}
+base.archivesName.set(projectDir.parentFile.name)
 
+// Used by muzzle
 val bootstrapRuntime by configurations.creating {
   isCanBeConsumed = false
   isCanBeResolved = true
 }
 
-afterEvaluate {
-  // TODO(anuraaga): There is no obvious reason this needs to be in afterEvaluate but it breaks
-  // Finatra tests. Fix it.
-  dependencies {
-    compileOnly(project(":instrumentation-api"))
-    compileOnly(project(":javaagent-api"))
-    compileOnly(project(":javaagent-bootstrap"))
-    // Apply common dependencies for instrumentation.
-    compileOnly(project(":javaagent-extension-api")) {
-      // OpenTelemetry SDK is not needed for compilation
-      exclude(group = "io.opentelemetry", module = "opentelemetry-sdk")
-      exclude(group = "io.opentelemetry", module = "opentelemetry-sdk-metrics")
-    }
-    compileOnly(project(":javaagent-tooling")) {
-      // OpenTelemetry SDK is not needed for compilation
-      exclude(group = "io.opentelemetry", module = "opentelemetry-sdk")
-      exclude(group = "io.opentelemetry", module = "opentelemetry-sdk-metrics")
-    }
-    compileOnly("net.bytebuddy:byte-buddy")
-    annotationProcessor("com.google.auto.service:auto-service")
-    compileOnly("com.google.auto.service:auto-service")
-    compileOnly("org.slf4j:slf4j-api")
-
-    testImplementation("io.opentelemetry:opentelemetry-api")
-
-    testImplementation(project(":testing-common"))
-    testAnnotationProcessor("net.bytebuddy:byte-buddy")
```

Comment:
Deleted this byte-buddy dependency while migrating since I couldn't find it being used